### ISSUE TITLE
JetBrains Rider like features

### ISSUE THREAD

**Issue by MordechaiHadad** [nvim-lua/wishlist#33]
**What?**
So I have made a list based on stuff i really enjoyed seeing in Rider:

- Context Action Menu via Alt + Enter
  - Could be a really useful feature first show code actions and then user mapped actions such as renaming etc instead of remembering a command/10 million keybinds, with the ability to search mapped actions

- Generate variable name suggestions if type is not a primitive type
- Move function/class/struct/etc to another file using treesitter(?)
- Alt + {button} for quick action
  - When cursor hovering on symbol function/class/variable do quick actions such as: quick delete/rename/move up/down and could be used for many other symbol specific actions

- symbol rearranging using something like CTRL(or ALT when GUI) + Up/Down/Left/Right when cursor hovering symbol

**Why?**
JetBrains rider and or more generally JetBrains IDEs have some of the best programmer-language supporting features which really does boost the efficiency of the programmer using whatever the programmer is using, and could be somewhat "easily" implemented considering we have LSPs and tree-sitter

**Potential existing implementations:**
Not that I could find.

**Potential pitfalls*:*
No idea to be honest

**Comment by Anrock** [comment]
> Context Action Menu via Alt + Enter
> Alt + {button} for quick action

Sounds like lighbulb/lspsaga 

>Move function/class/struct/etc to another file using treesitter(?)

That sounds more like a per-language lsp-specific feature since for this edit to be correct module exports/imports and other semantic things should modified too and treesitter just doesn't know that.

**Comment by MordechaiHadad** [comment]
> 
> 
> > Context Action Menu via Alt + Enter
> > Alt + {button} for quick action
> 
> Sounds like lighbulb/lspsaga

Yeah but that's only a very basic feature Context Menu by jetbrains has

>>Move function/class/struct/etc to another file using treesitter(?)
>
>That sounds more like a per-language lsp-specific feature since for this edit to be correct module exports/imports and other semantic things should modified too and treesitter just doesn't know that.

Yeah i agree.

**Comment by AnatoleLucet** [comment]
> Move function/class/struct/etc to another file using treesitter(?)

Maybe [refactoring.nvim](https://github.com/ThePrimeagen/refactoring.nvim)?
